cypress/integration/core-opensearch-dashboards/opensearch-dashboards/apps/query_enhancement/dataset_navigator.spec.js

```diff
@@ -0,0 +1,174 @@
+      cy.get(`[title="Indexes"]`).click();
+      cy.get(`[title="Default Cluster"]`).click();
+      cy.get(`[title="logstash-2015.09.20"]`).click();
+      cy.getElementByTestId('datasetSelectorNext').click();
+
+      cy.get(`[class="euiModalHeader__title"]`).should(
+        'contain',
+        'Step 2: Configure data'
+      );
+      // should have two options: SQL and PPL
+      cy.get('option').should('have.length', 2);
+      cy.getElementByTestId('advancedSelectorConfirmButton').click();
+
+      cy.waitForLoaderNewHeader();
+
+      // Selected language should be SQL
+      cy.getElementByTestId('queryEditorLanguageSelector').should(
+        'contain',
+        'SQL'
+      );
+
+      // The following steps are needed because when selecting SQL, discover loaded with data but the
+      // multi-line query editor are not loaded properly(it renders a single line query bar) unless we select SQL again
+      // This bug only exist in cypress test; can not reproduce manually
+      cy.get(`[data-test-subj="queryEditorLanguageSelector"]`).click();
+      cy.get(`[class~="languageSelector__menuItem"]`)
+        .should('have.length', 2)
+        .eq(1)
+        .click({
+          force: true,
+        });
+      cy.waitForLoaderNewHeader();
+      cy.get(`[data-test-subj="queryResultCompleteMsg"]`).should('be.visible');
+
+      // Switch language to PPL
+      cy.get(`[data-test-subj="queryEditorLanguageSelector"]`).click();
+      cy.get(`[class~="languageSelector__menuItem"]`).eq(0).click({
+        force: true,
+      });
+      cy.waitForLoaderNewHeader();
+      cy.get(`[data-test-subj="queryResultCompleteMsg"]`).should('be.visible');
+    });
+
+    it('with PPL as default language', function () {
+      cy.get(`[class~="datasetSelector__button"]`).click();
+      cy.get(`[class~="datasetSelector__advancedButton"]`).click();
+      cy.get(`[title="Indexes"]`).click();
+      cy.get(`[title="Default Cluster"]`).click();
+      cy.get(`[title="logstash-2015.09.21"]`).click();
+      cy.getElementByTestId('datasetSelectorNext').click();
+
+      cy.get(`[class="euiModalHeader__title"]`).should(
+        'contain',
+        'Step 2: Configure data'
+      );
+      // should have two options: SQL and PPL; PPL should be selected
+      cy.getElementByTestId('advancedSelectorTimeFieldSelect').select(
+        '@timestamp'
+      );
+      cy.getElementByTestId('advancedSelectorConfirmButton').click();
+
+      cy.waitForLoaderNewHeader();
+
+      // Selected language should be PPL
+      cy.getElementByTestId('queryEditorLanguageSelector').should(
+        'contain',
+        'PPL'
+      );
+
+      cy.waitForLoaderNewHeader();
+      cy.getElementByTestId('queryResultCompleteMsg').should('be.visible');
+      cy.getElementByTestId('queryEditorFooterTimestamp').should(
+        'contain',
+        '@timestamp'
+      );
+
+      // Switch language to SQL
+      cy.getElementByTestId('queryEditorLanguageSelector').click();
```

Comment:
could we wrap the switch language to a method to avoid duplication of code

Reply:
added a function for switch langauge